contracts/services/kyc/KYCVerifierV1.sol

```diff
@@ -10,34 +10,56 @@ import {
     DeploymentBlockNonInitializable
 } from "../../DeploymentBlockNonInitializable.sol";
 import {ERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import {EIP712} from "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
 
 /**
  * @title KYCVerifierV1
  * @author Decent Labs
- * @notice Mock implementation of KYC verification service
- * @dev This contract implements IKYCVerifierV1, providing a simple KYC
- * verification service that always returns true.
+ * @notice KYC verification service using EIP-712 signature verification
+ * @dev This contract implements IKYCVerifierV1, providing KYC verification
+ * through cryptographic signature verification.
  *
  * Implementation details:
- * - Mock implementation for testing and development
- * - Always returns true for any address verification
+ * - Uses EIP-712 structured data signing for verification
+ * - Requires signature from authorized verifier address
  * - Deployed as singleton service per chain
- * - Production implementations would integrate with real KYC providers
+ * - Supports operating contract-specific verification
  *
- * Production considerations:
- * - Real implementations would check on-chain attestations
- * - Could integrate with identity protocols or oracles
- * - May use merkle trees or signature verification
- * - Should implement proper access controls for updates
+ * Security considerations:
+ * - Verifier address is immutable and set at deployment
+ * - Uses ECDSA signature recovery for verification
+ * - EIP-712 prevents signature replay across different domains
+ * - Operating contract context prevents cross-contract signature reuse
  *
  * @custom:security-contact security@decentlabs.io
  */
 contract KYCVerifierV1 is
     IKYCVerifierV1,
     IVersion,
     DeploymentBlockNonInitializable,
-    ERC165
+    ERC165,
+    EIP712
 {
+    // ======================================================================
+    // STATE VARIABLES
+    // ======================================================================
+
+    address private immutable _verifier;
+
+    bytes32 internal constant TYPEHASH =
+        keccak256(
+            "VerificationData(address operatingContract,address account)"
+        );
```

Comment:
I think the signatures should include a `nonce` and an `expiry` timestamp, to prevent replay attacks (track used nonces in contract state), and to prevent stale signatures from being used.

Reply:
In this context, how would a replay attack be used? Also, what does a stale signature mean in this context?

I do think the concept of an `expiry` timestamp could make sense here, but not really seeing the use-case for a nonce

Reply:
For example, without a protection for replay attacks, someone could use the same signature for multiple `increaseCommitment` calls, even if their KYC status has been revoked (in the offchain system) after the first call.

Reply:
@adamgall Eddie said we should support revoking KYC status. Do you agree that just having the signature expiration timestamp can satisfy this? We can just always keep the expiration timestamp very short into the future - like a couple minutes or so.

Reply:
Why wouldn't you want to include a nonce as well? Feels like that is best practice. Is it the gas overhead?

Reply:
Ok, I'll add nonces and expiration timestamps

Reply:
this has been added, should be ready for a re-review